pandas/tests/indexing/test_iloc.py

```diff
@@ -1188,6 +1189,17 @@ def test_iloc_getitem_int_single_ea_block_view(self):
         arr[2] = arr[-1]
         assert ser[0] == arr[-1]
 
+    def test_iloc_setitem_multicolumn_to_datetime(self):
+
+        # GH#20511
+        df = DataFrame({"A": ["2022-01-01", "2022-01-02"], "B": ["2021", "2022"]})
+
+        df.iloc[:, [0]] = DataFrame({"A": to_datetime(["2021", "2022"])})
+        expected = DataFrame(
+            {"A": to_datetime(["2021", "2022"]), "B": ["2021", "2022"]}
+        )
+        tm.assert_frame_equal(df, expected, check_dtype=False)
```

Comment:
why is check_dtype=False?

Reply:
Because the datatype remains as **object** after assign back not converted to datetime 

```
>>> df = DataFrame({"A": ["2022-01-01", "2022-01-02"], "B": ["2021", "2022"]})
>>> 
>>> 
>>> df.iloc[:, [0]] = DataFrame({"A": to_datetime(["2021", "2022"])})
>>> 
>>> 
>>> df.dtypes
A    object
B    object
dtype: object
>>> 
>>> df
                     A     B
0  2021-01-01 00:00:00  2021
1  2022-01-01 00:00:00  2022
>>> 

```

Reply:
then pls construct the exact expected series. this may actually be a bug

Reply:
After the assignment if we check the dtypes it shows as object

```
In [89]: df = DataFrame({"A": ["2022-01-01", "2022-01-02"], "B": ["2021", "2022"]})

In [90]:  df.iloc[:, [0]] = df.iloc[:, [0]].apply(to_datetime)

In [91]: df.dtypes
Out[91]: 
A    object
B    object
dtype: object
```

But if we use iat to take a look at one of the single value which is converted it show as Timestamp

```
In [92]: df.iat[0,0]
Out[92]: Timestamp('2022-01-01 00:00:00')
```

So constructed the exact expected frame using Timestamp and dtype as object